From: Hannes Reinecke <hare@suse.de>
Date: Mon, 21 Jan 2013 11:55:05 +0100
Subject: dasd: Rename ioctls
Patch-Mainline: n/a
References: FATE#311379

As per request from IBM the ioctls should be named 'abortio'.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 arch/s390/include/asm/dasd.h    |    8 ++++----
 drivers/s390/block/dasd_ioctl.c |   16 ++++++++--------
 2 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/s390/include/asm/dasd.h b/arch/s390/include/asm/dasd.h
index 52cd02a..0a56f6f 100644
--- a/arch/s390/include/asm/dasd.h
+++ b/arch/s390/include/asm/dasd.h
@@ -262,10 +262,10 @@ struct dasd_snid_ioctl_data {
 #define BIODASDQUIESCE _IO(DASD_IOCTL_LETTER,6) 
 /* Resume IO on device */
 #define BIODASDRESUME  _IO(DASD_IOCTL_LETTER,7) 
-/* IO timeout on device */
-#define BIODASDTIMEOUT _IO(DASD_IOCTL_LETTER,240) 
-/* Resync IO on device */
-#define BIODASDRESYNC  _IO(DASD_IOCTL_LETTER,241) 
+/* Abort all IO on a device */
+#define BIODASDABORTIO _IO(DASD_IOCTL_LETTER,240) 
+/* Allow IO on a device */
+#define BIODASDALLOWIO  _IO(DASD_IOCTL_LETTER,241) 
 
 
 /* retrieve API version number */
diff --git a/drivers/s390/block/dasd_ioctl.c b/drivers/s390/block/dasd_ioctl.c
index 48973fc..4979b31 100644
--- a/drivers/s390/block/dasd_ioctl.c
+++ b/drivers/s390/block/dasd_ioctl.c
@@ -141,9 +141,9 @@ static int dasd_ioctl_resume(struct dasd_block *block)
 }
 
 /*
- * Timeout device.
+ * Abort all failfast I/O on a device
  */
-static int dasd_ioctl_timeout(struct dasd_block *block)
+static int dasd_ioctl_abortio(struct dasd_block *block)
 {
 	unsigned long flags;
 	struct dasd_device *base;
@@ -183,9 +183,9 @@ static int dasd_ioctl_timeout(struct dasd_block *block)
 }
 
 /*
- * Resync a device
+ * Allow I/O on a device
  */
-static int dasd_ioctl_resync(struct dasd_block *block)
+static int dasd_ioctl_allowio(struct dasd_block *block)
 {
 	unsigned long flags;
 	struct dasd_device *base;
@@ -493,11 +493,11 @@ int dasd_ioctl(struct block_device *bdev, fmode_t mode,
 	case BIODASDRESUME:
 		rc = dasd_ioctl_resume(block);
 		break;
-	case BIODASDTIMEOUT:
-		rc = dasd_ioctl_timeout(block);
+	case BIODASDABORTIO:
+		rc = dasd_ioctl_abortio(block);
 		break;
-	case BIODASDRESYNC:
-		rc = dasd_ioctl_resync(block);
+	case BIODASDALLOWIO:
+		rc = dasd_ioctl_allowio(block);
 		break;
 	case BIODASDFMT:
 		rc = dasd_ioctl_format(bdev, argp);
-- 
1.7.4.2

